rqt_py_common/src/rqt_py_common/rqt_roscomm_util.py

```diff
@@ -112,22 +119,28 @@ def iterate_packages(subdir):
         @type subdir: str
         @raise ValueError:
         """
-        if subdir == None or subdir == '':
-            raise ValueError('Invalid package subdir = {}'.format(subdir))
+        # if subdir is None or subdir == '':
+        #     raise ValueError('Invalid package subdir = {}'.format(subdir))
 
-        rospack = rospkg.RosPack()
+        # rospack = rospkg.RosPack()
+        # pkgs = rospack.list()
 
-        pkgs = rospack.list()
-        rospy.logdebug('pkgs={}'.format(pkgs))
-        for p in pkgs:
-            d = os.path.join(rospack.get_path(p), subdir)
-            rospy.logdebug('rospack dir={}'.format(d))
-            if os.path.isdir(d):
-                yield p, d
+        # RqtRoscommUtil._logger.debug('pkgs={}'.format(pkgs))
+        # for p in pkgs:
+        #     d = os.path.join(rospack.get_path(p), subdir)
+        #     RqtRoscommUtil._logger.debug('rospack dir={}'.format(d))
+        #     if os.path.isdir(d):
+        #         yield p, d
+        RqtRoscommUtil._logger.error("iterate_packages: not implemented in ROS2))")
 
     @staticmethod
     def list_files(package, subdir, file_extension='.launch'):
         """
+        Note: Mlautman 11/2/2018
```

Comment:
Shouldn't there be a function like this to find launch files and any other necessary ros2 files for a package-specific subdirectory?

Reply:
implemented